Data/models/Device.swift

```diff
@@ -49,10 +51,26 @@ public final class Device: NSManagedObject, Syncable, CRUD {
         
         // No guard, let bleed through to allow 'empty' devices (e.g. local)
         let root = root as? SyncDevice
+        
+        let syncUUID = root?.objectId ?? SyncCrypto.uniqueSerialBytes(count: 16)
+        
+        // There can't be more than two device with the same syncUUID.
+        // A race condition could sometimes happen while getting two `Sync.resolvedSyncRecords` callbacks at the same time.
+        // Fixing issue #692 should help with it, until then we do a simple guard and disallow adding another device
+        // with the same syncUUID.
+        if let syncDisplayUUID = SyncHelpers.syncDisplay(fromUUID: syncUUID) {
+            let predicate = NSPredicate(format: "syncDisplayUUID == %@", syncDisplayUUID)
+            let first = Device.first(where: predicate, context: context)
+            if first != nil {
+                log.warning("A device with sync UUID \(syncDisplayUUID) already exists. Aborting inserting a new device.")
+                return nil
```

Comment:
What do you think about just failing silently and letting this bleed through?

```
var device = Device.first(where: predicate, context: context)
if device == nil {
  // this is expected
  device = Device(entity: Device.entity(context: context), insertInto: context)
  device.created = root?.syncNativeTimestamp ?? Date()
  device.syncUUID = syncUUID
}

device.update(syncRecord: root)
```

Reply:
The question is, do we want to eventually update record on insert in this case?
We do something similar in `Sync.resolvedRecords` so that shouldn't be a problem